### PR TITLE
Fix: Completing goal twice adds it to archive twice

### PR DESCRIPTION
**Pull request by fendermany**
Fix #875

### CODE REVIEW

**Comment by tijlleenders** [comment]
Looking a lot better @fendermany !
An explanation of the problem and fix would be nice in the PR description.
You can ignore the Cypress tests - we need to update those....

@Tushar-4781 is it sensible for Dmitry to continue working on this PR given the changes we discussed today?
If not, can you propose some other issue for him to work on?

**Comment by Tushar-4781** [comment]
@fendermany have a look on #877

File: src/api/GoalsAPI/index.ts [hunk 1]
```diff
@@ -124,11 +111,17 @@ export const updateGoal = async (id: string, changes: object) => {
 
 export const archiveGoal = async (goal: GoalItem) => {
   const updatedGoalStatus = { status: 1 };
+  const parentGoal = await getGoal(goal.parentGoalId);
+  
+  if (parentGoal && parentGoal.collaboration.status !== 'none') {
```
Comment (Tushar-4781) [inline src/api/GoalsAPI/index.ts:116]: Apologies, there are some modifications in the schema of goalItem as our story changed a bit. but approving for now

but Use **collaboration.status === "accept"** as that's the case where we will be following the exception

File: src/components/GoalsComponents/GoalSublistPage/GoalSublistPage.tsx [hunk 2]
```diff
@@ -33,7 +41,7 @@ export const GoalSublist = () => {
   const callResetHistory = useSetRecoilState(resetGoalsHistory);
   const setShowAddGoalOptions = useSetRecoilState(displayAddGoalOptions);
 
-  const [tapCount, setTapCount] = useState(defaultTap);
+  const [tapCount, setTapCount] = useState<{ open: string | number, click: number }>(defaultTap);
```
Comment (Tushar-4781) [inline src/components/GoalsComponents/GoalSublistPage/GoalSublistPage.tsx:44]: Please do not engage yourself in refactoring. this creates conflicts in coming PR(s). Refactoring is done only when there is no **feature** in progress.

File: src/components/GoalsComponents/GoalSublistPage/GoalSublistPage.tsx [hunk 3]
```diff
@@ -51,25 +59,21 @@ export const GoalSublist = () => {
...
 
-  const archiveMyGoal = async (goal: GoalItem) => {
-    await archiveUserGoal(goal);
-    await getChildrenGoals(goalID).then((fetchedGoals) => setChildrenGoals(fetchedGoals));
-  };
   const removeChildrenGoal = async (goalId: string) => {
     if (parentGoal?.sublist) {
       // delete subgoals of this goal
-      removeChildrenGoals(goalId);
+      await removeChildrenGoals(goalId);
```
Comment (Tushar-4781) [inline src/components/GoalsComponents/GoalSublistPage/GoalSublistPage.tsx:76]: we don't need to wait for this. as we are deleting the root goal in the next line, the sublist will be unreachable anyhow. so please do not use await here here

**Review by Tushar-4781 — CHANGES_REQUESTED** [review]
@tijlleenders can you please check if the objective is achieved from UI point of view. Post that i will review rest of the files.

**Comment by tijlleenders** [comment]
Did not test for shared or collaborated goal.
Think @Tushar-4781 can check this better.

For 'normal' (sub)goal the new behavior is quite nice from UX perspective!

The archived goal disappears from the list when you add it and can only appear once in the list with the same name. If you archive twice with the same title & parent - the last archived overrides the archived version with identical title&parent.

**Pull request closed without merging**